src/main/java/io/anserini/qa/RetrieveSentences.java

```diff
@@ -0,0 +1,192 @@
+    public int hits = 100;
+  }
+
+  private static final Logger LOG = LogManager.getLogger(SearchWebCollection.class);
+
+  public Map<String, Double> computeIDF(QAargs qAargs) throws Exception {
+    IndexUtils util = new IndexUtils(qAargs.index);
+    FSDirectory directory = FSDirectory.open(new File(qAargs.index).toPath());
+    DirectoryReader reader = DirectoryReader.open(directory);
+
+    EnglishAnalyzer ea = new EnglishAnalyzer(CharArraySet.EMPTY_SET);
+    QueryParser qp = new QueryParser(LuceneDocumentGenerator.FIELD_BODY, ea);
+
+    Map<String, Double> sentenceIDF = new HashMap();
+    ClassicSimilarity similarity = new ClassicSimilarity();
+
+    //for each document in the ranked list, dump sentences
+    try (BufferedReader br = new BufferedReader(new FileReader(qAargs.output))) {
+      String line;
+
+      while ((line = br.readLine()) != null) {
+        String docid = line.trim().split(" ")[2];
+        List<Sentence> sentences = util.getSentDocument(docid);
+        for (Sentence sent: sentences) {
+          double idf = 0.0;
+          String[] terms = sent.text().split(" ");
+          for (String term: terms) {
+            try {
+              TermQuery q = (TermQuery) qp.parse(term);
+              Term t = q.getTerm();
+              idf += similarity.idf(reader.docFreq(t), reader.numDocs());
+            } catch (Exception e){
+              continue;
+            }
+          }
+          sentenceIDF.put(sent.text(), new Double(idf));
+        }
+      }
+    }
+    return sentenceIDF;
+  }
+
+  public SortedMap<Integer, String> readTopicsFile(String topicsFile) throws IOException {
+    SortedMap<Integer, String> map = new TreeMap<>();
+    String pattern = "<QApairs id=\'(.*)\'>";
+    Pattern r = Pattern.compile(pattern);
+
+    try (BufferedReader br = new BufferedReader(new FileReader(topicsFile))) {
+      String line;
+      String prevLine = "";
+
+      while ((line = br.readLine()) != null) {
+        Matcher m = r.matcher(line);
+        String id = "";
+        if (m.find()) {
+          id = m.group(1);
+        }
+
+        if (prevLine != null && prevLine.startsWith("<question>")) {
+          map.put(Integers.parseInt(id), line);
+        }
+        prevLine = line;
+      }
+    }
+    return map;
+  }
+
+  public void retrieveDocuments(QAargs qaArgs) throws Exception {
+    Path indexPath = Paths.get(qaArgs.index);
+    if (!Files.exists(indexPath) || !Files.isDirectory(indexPath) || !Files.isReadable(indexPath)) {
+      throw new IllegalArgumentException(qaArgs.index + " does not exist or is not a directory.");
+    }
+
+    IndexReader reader = DirectoryReader.open(FSDirectory.open(indexPath));
+
+    Directory dir;
+    dir = FSDirectory.open(Paths.get(qaArgs.index));
+
+    //using BM25 scoring model
+    Similarity similarity = null;
+    similarity = new BM25Similarity(0.9f, 0.4f);
+
+    RerankerCascade cascade = new RerankerCascade();
```

Comment:
Don't depend on `SearchWebCollection`, just copy out the lines you need. It's find to have a separate code path right now.